platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/DirectTaskScheduler.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.schedulers;
+
+import com.swirlds.common.metrics.extensions.FractionalTimer;
+import com.swirlds.common.wiring.TaskScheduler;
+import com.swirlds.common.wiring.WiringModel;
+import com.swirlds.common.wiring.builders.TaskSchedulerType;
+import com.swirlds.common.wiring.counters.ObjectCounter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * A scheduler that performs work immediately on the caller's thread.
+ *
+ * @param <OUT> the output type of the scheduler (use {@link Void} for a task scheduler with no output type)
+ */
+public class DirectTaskScheduler<OUT> extends TaskScheduler<OUT> {
+
+    private final UncaughtExceptionHandler uncaughtExceptionHandler;
+    private final ObjectCounter onRamp;
+    private final ObjectCounter offRamp;
+    private final FractionalTimer busyTimer;
+
+    /**
+     * Constructor.
+     *
+     * @param model                    the wiring model containing this task scheduler
+     * @param name                     the name of the task scheduler
+     * @param uncaughtExceptionHandler the uncaught exception handler
+     * @param onRamp                   an object counter that is incremented when data is added to the task scheduler
+     * @param offRamp                  an object counter that is decremented when data is removed from the task
+     * @param busyTimer                a timer that tracks the amount of time the task scheduler is busy
+     * @param stateless                true if the work scheduled by this object is stateless
+     */
+    public DirectTaskScheduler(
+            @NonNull final WiringModel model,
+            @NonNull final String name,
+            @NonNull final UncaughtExceptionHandler uncaughtExceptionHandler,
+            @NonNull final ObjectCounter onRamp,
+            @NonNull final ObjectCounter offRamp,
+            @NonNull final FractionalTimer busyTimer,
+            final boolean stateless) {
+        super(model, name, stateless ? TaskSchedulerType.DIRECT_STATELESS : TaskSchedulerType.DIRECT, false, true);
+
+        this.uncaughtExceptionHandler = Objects.requireNonNull(uncaughtExceptionHandler);
+        this.onRamp = Objects.requireNonNull(onRamp);
+        this.offRamp = Objects.requireNonNull(offRamp);
+        this.busyTimer = Objects.requireNonNull(busyTimer);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getUnprocessedTaskCount() {
+        return -1;
```

Comment:
why `-1`? use a constant

Reply:
Constant is now defined in `ObjectCounter`, it's called `COUNT_UNDEFINED`.